### ISSUE TITLE
ESP32 port for OTA

### ISSUE THREAD

**Issue by ajlennon** [aws/aws-iot-device-sdk-embedded-C#1693]
I am trying to support OTA with a Platform.io build environment...

After some investigating I have found platformio/platform-espressif32

ref: https://github.com/platformio/platform-espressif32

This contains a demo, espidf-aws-iot  which shows pub/sub and seemingly shadowing.
 
https://github.com/espressif/esp-aws-iot

This leverages v3.0.1 of this aws-iot-device-sdk-embedded-C and has a ports/* folder with what appear to be the needed function implementations. The pub/sub with AWS IoT works fine in this configuration.

The v3.0.1 port files are here: 

https://github.com/platformio/platform-espressif32/tree/develop/examples/espidf-aws-iot/components/esp-aws-iot/port

Next I am trying to support OTA. The latest release of the Embedded C SDK "202103.00" seems to include support for OTA.

However this seems to have been refactored extensively and the existing port for ESP32 for v3.0.1 doesn't seem to fit into the new architecture.

I have also seen there seems to be some kind of OTA support built into Amazon FreeRTOS for ESP32 but this doesn't fit into my requirement for a Platform.io build toolchain.

So.... can somebody tell me if there is an existing port for ESP32 for 202103.00 which I can leverage? 

Or can somebody tell me where to begin to migrate ports/* as I am finding it hard to follow the porting and migration documents I am finding, in particular to migrate from the v3.0.1 port/* files to 202103.00?

Thanks! Alex

**Comment by pvyawaha** [comment]
Hello,

Please take a look at the port in

**Comment by FanuelConrad** [comment]
Hello,
I'd like to if it's possible to perform OTA Update on ESP32 via AWS IoT and Platform.IO
Kindly show some guidance on the same
